src/server/src/routes/job.ts

```diff
@@ -0,0 +1,148 @@
+        expirationDate,
+        startDate,
+        endDate,
+        type,
+        title,
+        status,
+        minSalary,
+        maxSalary,
+        departmentId } = job;
+    if (!job) {
+        return res.status(BAD_REQUEST).json({
+            error: errors.paramMissingError,
+        });
+    }
+    try {
+        // Check if required field is missing.
+        if (!targetYears || !hoursPerWeek || !description || !startDate || !type
+            || !title || !status || !minSalary || !departmentId) {
+            return res.status(BAD_REQUEST).json({
+                error: errors.paramMissingError,
+            });
+        }
+
+        await createJob(targetYears,
+                        hoursPerWeek,
+                        description,
+                        expirationDate,
+                        startDate,
+                        endDate,
+                        type,
+                        title,
+                        status,
+                        minSalary,
+                        maxSalary,
+                        departmentId);
+        return res.status(CREATED).end();
+    } catch (error) {
+        logger.err(error);
+        return res
+            .status(INTERNAL_SERVER_ERROR)
+            .json(errors.internalServerError)
+            .end();
+    }
+});
+
+/******************************************************************************
+ *             POST Request example - Update - "POST /api/job/update"
+ ******************************************************************************/
+
+router.post('/update', async (req: jobRequest, res: Response) => {
+    const { job } = req.body;
+    const { targetYears,
+        hoursPerWeek,
+        description,
+        expirationDate,
+        startDate,
+        endDate,
+        type,
+        title,
+        status,
+        minSalary,
+        maxSalary,
+        departmentId,
+        id } = job;
+    if (!job) {
+        return res.status(BAD_REQUEST).json({
+            error: errors.paramMissingError,
+        });
+    }
+    try {
+        await updateJob(
+            targetYears,
+            hoursPerWeek,
+            description,
+            expirationDate,
+            startDate,
+            endDate,
+            type,
+            title,
+            status,
+            minSalary,
+            maxSalary,
+            departmentId,
+            id);
+        return res.status(OK).end();
+    } catch (error) {
+        logger.err(error);
+        return res
+            .status(INTERNAL_SERVER_ERROR)
+            .json(errors.internalServerError)
+            .end();
+    }
+});
+
+/******************************************************************************
+ *        DELETE Request example - Delete - "DELETE /api/job/delete/:id"
+ ******************************************************************************/
+
+router.delete('/delete/:id', async (req: jobRequest, res: Response) => {
+    const { id } = req.params;
```

Comment:
Add validation. If id is null then return an error response

Reply:
Done.